src/test/java/umk/mat/pajda/ProjektZespolowy/PatternPasswordValidatorTest.java

```diff
@@ -0,0 +1,49 @@
+package umk.mat.pajda.ProjektZespolowy;
+
+import jakarta.validation.ConstraintValidatorContext;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.context.ActiveProfiles;
+import umk.mat.pajda.ProjektZespolowy.validators.PatternPasswordValidator;
+
+@ActiveProfiles("tests")
+public class PatternPasswordValidatorTest {
+
+  private PatternPasswordValidator patternPasswordValidator;
+
+  @Mock private ConstraintValidatorContext mock;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.openMocks(this);
+    patternPasswordValidator = new PatternPasswordValidator();
+  }
+
+  @Test
+  public void noUppercaseTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("n1!", mock));
+  }
+
+  @Test
+  public void noLowercaseTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("N1!", mock));
+  }
+
+  @Test
+  public void noDigitTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("Nn!", mock));
+  }
+
+  @Test
+  public void noSpecialCharacterTest() {
+    Assertions.assertFalse(patternPasswordValidator.isValid("Nn1", mock));
+  }
+
+  @Test
+  public void succesPatternTest() {
+    Assertions.assertTrue(patternPasswordValidator.isValid("Nn1!", mock));
+  }
```

Comment:
does that mean that we accept 4 characters for password? I would say the bare minimum is 6 🤔

Reply:
Size validator contains length of password

Reply:
can we apply all validations in the DTO?, this pattern validator seems to be a workaround for lack of features in the dto validation library 🤔

Reply:
something like 
@size()
@matches(`regex here`)

Reply:
I'm not sure if  that's exactly what you mean,  but in Spring, there's a default pattern validator that I use for the surname and name fields. However, when I tried to apply it to the password field, but it didn't work as expected. So I created a custom validator, wchich check the password contains at least 1 digit, 1 uppercase letter, 1 lowercase letter, and 1 special character.

Reply:
i changed it and now it works with this default pattern